packages/lodestar/src/api/impl/beacon/beacon.ts

```diff
@@ -93,9 +93,8 @@ export class BeaconApi implements IBeaconApi {
 
   public getBlockStream(): AsyncIterable<SignedBeaconBlock> {
     return new EventIterator<SignedBeaconBlock>((push) => {
-      this.chain.on("processedBlock", (block) => {
-        push(block);
-      });
+      this.chain.on("processedBlock", push);
+      return () => this.chain.off("processedBlock", push);
```

Comment:
this won't be called unless you call stop (second argument in constructor which is awkward.

Reply:
Maybe we can update to event-iterator 2.0.0 which seems to have a simpler API https://www.npmjs.com/package/event-iterator#advanced-examples

Reply:
Yeah sure ( I think we have fork endpoint with this as well). I just briefly checked, but not sure if simpler api helps you.
It would be ideal if EventIntertor class would expose stop method. Maybe you can extend it with that method?
In endpoint implementation, you would get it and call `req.raw.on("close", () => lodestarEventIterator.stop())`

Reply:
So you mean that it's `fastify-sse-v2` how is not handling the close event properly? Note that the plugin is adding types and headers by itself https://github.com/NodeFactoryIo/fastify-sse-v2/blob/6a91fedb4a4aa82d1fd13f6cb14909cc028d8ed5/src/plugin.ts#L14 so this redundancy may be cause?

Reply:
plugin doesn't have access to req so it cannot detect req closing, Once request is closed, fastify closes stream, but I cannot notify event-iterator to stop as I only get async iterable

Reply:
https://github.com/NodeFactoryIo/fastify-sse-v2/issues/2